sycl/source/detail/graph_impl.cpp

```diff
@@ -1236,18 +1237,22 @@ void exec_graph_impl::update(
           sycl::make_error_code(errc::invalid),
           "Node passed to update() is not part of the graph.");
     }
-    if (Node->MCGType != sycl::detail::CG::Kernel) {
-      throw sycl::exception(errc::invalid, "Cannot update non-kernel nodes");
-    }
 
-    if (Node->MCommandGroup->getRequirements().size() == 0) {
-      continue;
+    if (!(Node->isEmpty() || Node->MCGType == sycl::detail::CG::Kernel ||
+          Node->MCGType == sycl::detail::CG::Barrier)) {
+      throw sycl::exception(errc::invalid,
+                            "Cannot update node type. Node must be be of "
```

Comment:
I think the exception message could be a bit more clear. Something like "Unsupported node type for update. Only kernel, barrier and empty nodes are supported."